provider.tf

```diff
@@ -1,3 +1,8 @@
-provider "azurerm" {
-  features {}
+terraform {
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = ">= 2.35.0"
```

Comment:
why are you forcing to use a version >= to 2.35.0 here ?

Reply:
it is the version that the code was tested with. and it will allow the user to use newer versions if they want.

Reply:
I can change it or remove it if you would like.

Reply:
I am currently working at migrating the project/CI to Terraform 0.14.4 as it was still using TF 0.12.x.
I will require a provider >= 2.0, I don't see any restriction to not support all 2.X version here. So yes, you will probably be able to remove that part from the PR, after you sync your fork with the new updates from #14

Reply:
Ok, once #14 has been merged I will update and remove the 2.35 requirement.

Reply:
@mozts2005 thanks for your patience. #14 is now merge. The requirement for the provider has been completely removed at the same time, so it can be done outside of the module, by the caller. Please update your fork and do the few changes we discussed previously.
I also fixed the CI, that must be triggered when you will commit your changes. Once everything green I will be able to approve your PR. LMK if you have any question!
Thanks!

Reply:
pr updated